encrypted-media-respec.html

```diff
@@ -605,8 +605,112 @@
 
           <section>
             <h5>Get Supported Configuration</h5>
-            <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>, and <var>origin</var>, this algorithm returns a supported configuration or <code>null</code> as appropriate.</p>
-            <p class="note">Unrecognized dictionary members in <var>candidate configuration</var> are ignored per [[WebIDL]] and will never reach this algorithm. Thus, they cannot be considered as part of the configuration.</p>
+            <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>, and <var>origin</var>, this algorithm returns a supported configuration or <code>NotSupported</code> as appropriate.</p>
+            <p class="note">Unrecognized dictionary members in <var>candidate configuration</var> are ignored per [[WebIDL]] and will never reach this algorithm. Thus, they cannot be considered as part of the configuration.
+            </p>
+            <div class="note">
+              <p>
+              For certain configurations, it may be required to obtain user consent or inform the user. User Agents have some flexibility to determine
+              whether consent is required for a specific configuration and whether such consent may also apply to other configurations. For example,
+              consent to one configuration may also imply consent for less powerful, more restricted configurations. Equally, a denial of consent for
+              one configuration may imply denial of consent for more powerful, less restricted configurations.
+              </p>
+              <p>
+              Supported configurations, including supported audio and video codecs, may depend on availability of optional capabilities such as
+              <a def-id="distinctive-identifier"></a>s and persistent state. The following algorithm iteratively tries configurations with different
+              requirement values for these optional configuration elements until consent is achieved.
+              </p>
+              <p>
+              User Agents should persist consent responses at least for the duration of the <a def-id="requestMediaKeySystemAccess"></a>
+              algorithm in order to avoid repeated requests to the user for similar configurations.
+              </p>
+            </div>
+            <ol>
+              <li>
+                <p>Let <var>requirements</var> be the list [ <code>distinctiveIdentifier</code>, <code>persistentState</code> ].</p>
```

Comment:
Yes, that's better.

To check we're on the same page, the changes will be:
- remove the construction of "all possible combinations"
- change the loop to "repeat until done" in some form
- Get Supported Configurations and Consent will return a _restrictions_ list with the `ConsentDenied` return value. The _restrictions_ list is a list of those elements of the _modified configuration_ of type `MediaKeyRequirement` which had value `optional` and for which consent has been denied.
- If _restrictions_ is non-empty, update _modified candidate configuration_ to set all the members listed in _restrictions_ to `not-allowed` and continue the loop
- If _restrictions_ is empty, it means none of the `MediaKeyRequirement` elements were optional and consent was denied anyway, so we return `NotSupported`.

However, we might have a different problem, unrelated to the handling of these `optional` values that could affect this section: suppose the site sets DI to `required` and asks for some set of codecs with hardware DRM or a different set of codecs with software DRM. And now suppose the software DI implementation supports all the recommendations of the security section (and so requires the user to be informed, but not consent) but the HW one does not. The user denies consent for DI with HW DRM, but the SW option is still viable. I don't believe we can handle this with the existing text.